includes/admin/wc-admin-functions.php

```diff
@@ -214,21 +214,27 @@ function wc_maybe_adjust_line_item_product_stock( $item, $item_quantity = -1 ) {
 		return false;
 	}
 
-	$diff = $item_quantity - $already_reduced_stock;
+	$order                  = $item->get_order();
+	$refunded_item_quantity = $order->get_qty_refunded_for_item( $item->get_id() );
+	$diff                   = $item_quantity + $refunded_item_quantity - $already_reduced_stock;
 
 	if ( $diff < 0 ) {
 		$new_stock = wc_update_product_stock( $product, $diff * -1, 'increase' );
-	} else {
+	} elseif ( $diff > 0 ) {
 		$new_stock = wc_update_product_stock( $product, $diff, 'decrease' );
 	}
 
 	if ( is_wp_error( $new_stock ) ) {
 		return $new_stock;
 	}
 
-	$item->update_meta_data( '_reduced_stock', $item_quantity );
+	$item->update_meta_data( '_reduced_stock', $item_quantity + $refunded_item_quantity );
 	$item->save();
 
+	if ( 0 === $diff ) {
```

Comment:
Perhaps better to move this above the update meta call as if there is no difference no stock should be adjusted?

Reply:
you still need to adjust `_reduced_stock` in line 231, so we cannot move it above (even if diff is 0).

Reply:
Not sure I understand why it needs to adjust the meta value when no stock adjustment was made? You have the following statement which does the actual stock adjustment

```
if ( $diff < 0 ) {
		$new_stock = wc_update_product_stock( $product, $diff * -1, 'increase' );
	} elseif ( $diff > 0 ) {
		$new_stock = wc_update_product_stock( $product, $diff, 'decrease' );
	}
```
If $diff is 0 not adjust is made, yet you are updating the _reduced_stock meta value even if no adjustment was made.

Reply:
Ah you are right, I am presuming that we were updating `_reduced_stock` value in order to maintain consistency, but you don't need to when the diff is 0. 

Updating to return early when diff is 0.